### PR TITLE
Get SonarCube tracking net code from latest release (unstable branch)

### LINKED ISSUE #2058: SonarCloud analysis run on PR branch has wrong `coverage on New Code`
Refer: https://github.com/apache/kvrocks/pull/2055/checks?check_run_id=20897911202

### PR DESCRIPTION
**Pull request by aleksraiden**
Add a latest release branch as sonar.newCode.referenceBranch to track a new code, added after latest release. Closes #2058, #2059

### CODE REVIEW

File: .github/workflows/sonar.yaml [hunk 1]
```diff
@@ -81,4 +81,5 @@ jobs:
             --define sonar.projectKey=apache_kvrocks \
             --define sonar.organization=apache \
             --define sonar.scm.revision=${{ github.event.workflow_run.head_sha }} \
-            --define sonar.pullrequest.key=$PR_NUMBER
+            --define sonar.pullrequest.key=$PR_NUMBER \
+            --define sonar.newCode.referenceBranch=2.8
```
Comment (git-hulk) [inline .github/workflows/sonar.yaml]: We should use the unstable branch instead of 2.8 since all new updates are merged into the unstable branch.

Comment (aleksraiden) [reply .github/workflows/sonar.yaml]: Done

**Comment by jihuayu** [comment]
I am testing this modification on my test project to see if it takes effect. This will take some time, please wait.

**Comment by jihuayu** [comment]
I have a try, but it doesn't work in my repo. Does it work in your repo? @aleksraiden 

Following the doc, It seems like the `sonar.newCode.referenceBranch` option is only supported by SonarQube. In SonarCloud we don't have this option.

SonarCloud only version and Number of days.
Maybe we can use the `Specific date` option, and use the last 30 days' code as new code in sonarcloud setting. @git-hulk 
This option can fixed https://github.com/apache/kvrocks/issues/2059.

And the https://github.com/apache/kvrocks/issues/2058 is another problem. I am trying it in [my repo](https://github.com/jihuayu/kvrocks-test/pull/2).

## Refer:
- [the new code options for SonarQube](https://docs.sonarsource.com/sonarqube/latest/project-administration/clean-as-you-code-settings/defining-new-code/#new-code-definition-options)
- [the new code options for SonarCloud](https://docs.sonarsource.com/sonarcloud/improving/new-code-definition/#new-code-definition-options)
- [the SonarCloud PR example](https://docs.sonarsource.com/sonarcloud/advanced-setup/ci-based-analysis/other-cis/)

**Comment by aleksraiden** [comment]
@jihuayu thanks about your investigation, you are absolutely correct, I see a docs about sonar cube, not a cloud version.

**Comment by jihuayu** [comment]
@aleksraiden 
Thank you for your help. 
I am looking into this issue. Just a few more questions to deal.

**Comment by PragmaTwice** [comment]
We are cleaning branches of this repo and this branch seems stale for a long time.

I'll close it and delete this branch. Feel free to submit a new patch if you want.

Also, if you want to keep these branches for a long time, you can push them into your own fork.

**Pull request closed without merging**